blocks/product-details/product-details.js

```diff
@@ -91,9 +91,17 @@ class ProductDetailPage extends Component {
   }
 
   async componentDidMount() {
-    const product = await getProduct(getSkuFromUrl());
-
-    if (!product) {
+    let product;
+    try {
+      if (window.getProductPromise) {
```

Comment:
I think this will work. It might be a bit brittle though, since the PDP block needs to be loaded before any other block. Otherwise `window.getProductPromise` might be `undefined` if the preload did not work.

Maybe a better alternative and something the product recommendations block is already using, is to read the data from ACDL. This has the advantage that you can register an event listener at any time and it is triggered even if the product data was provided already. See https://github.com/hlxsites/aem-boilerplate-commerce/blob/main/blocks/product-recommendations/product-recommendations.js#L210.